src/Orleans/Telemetry/Consumers/FileTelemetryConsumer.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.IO;
+
+namespace Orleans.Runtime
+{
+    public class FileTelemetryConsumer : ITraceTelemetryConsumer
+    {
+        private StreamWriter _logOutput;
+        private readonly object _lockObj = new object();
+
+        public FileTelemetryConsumer(string fileName) : this(new FileInfo(fileName))
+        {
+        }
+
+        public FileTelemetryConsumer(FileInfo file)
+        {
+            var fileExists = File.Exists(file.FullName);
+            _logOutput = fileExists ? file.AppendText() : file.CreateText();
+            file.Refresh();
+        }
+
+        public void TrackTrace(string message)
+        {
+            lock (_lockObj)
+            {
+                if (_logOutput == null) return;
+                _logOutput.WriteLine(message);
+                _logOutput.Flush();
```

Comment:
We can enable AutoFlush. The old `LogWriterToFile` does it and it would be prudent IMHO to keep that way. Many tools keep the file being read while it is written to provide real-time updates for example, if we are talking about a Linux silo that someone will be calling `tail -f logfile.log` it will be reading every single new line and printing on a terminal as they are written. Since we are talking about logs and not a massive hot path, I won't bother and would keep that way instead of add the complexity to manage how to call call flush and loose the real-time tail of events. Note that most of people only use files as a debugging thing if they don't have other way to do it. People prefer to have events on a more searchable way. What you think on keep it and look at the perftests to see if it will make any degradation?